source/Projectile.h

```diff
@@ -47,7 +48,7 @@ class Projectile : public Body {
 	};
 
 public:
-	Projectile(const Ship &parent, Point position, Angle angle, const Weapon *weapon);
+	Projectile(const Ship &parent, Point position, Angle angle, const Hardpoint *hardpoint, const Weapon *weapon);
```

Comment:
Hardpoint scode is necessary so you can call `Hardpoint::SpinupProgress()` in `Projectile::ApplyInaccuracy(Hardpoint)`.  Side note, I tried to add these methods once before in #6752 but tehhowch (rightly) rejected them they weren't really relevant at all.  They are more relevant here but since they need to be overloaded depending on whether the primary munition calls it or not makes them a bit redundant.  They are much cleaner though.

Reply:
It'd be simpler to just pass the SpinupProgress as a variable where necessary instead of passing the whole Hardpoint.

Reply:
Done, good point.